### ISSUE TITLE
Bug: import/first conflicts with Vue3 <script> and <script setup>

### ISSUE THREAD

**Issue by wsw70** [eslint/eslint#15126]
### Environment

Node version: v14.15.4
npm version: v7.15.1
Local ESLint version: v7.32.0
Global ESLint version: v7.31.0 (Currently used)
Operating System: win32 10.0.19043

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

A Vue3 (3.2+) component with a <script> and <script setup> section

</details>

```js
<script>
  const a = 3
</script>

<script setup>
  import { DateTime } from 'luxon'
</script>
```

### What did you expect to happen?

Nothing, the code is correct.

### What actually happened?

The error

     ESLint: Import in body of module; reorder to top. (import/first)

is displayed. The solution is to move the `import` at the top of `<script>` but this breaks the logic of importing in the proper `<script>` or `<script setup>`  and generates warnings about unused imports.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
This error is coming from the eslint-plugin-import package. You’ll need to open an issue with them if you don’t agree with how the rule is working.
